Sources/MacroToolkit/EnumRawRepresentableType.swift

```diff
@@ -0,0 +1,21 @@
+import SwiftSyntax
+
+/// Enum raw values can be strings, characters, or any of the integer or floating-point number types.
+public enum EnumRawRepresentableType {
+    case string
+    case character
+    case number
```

Comment:
I feel like `number` should be split into `integer` and `floatingPoint` given that if someone is trying to parse out the enum's raw values, defaulting to either `IntegerLiteral` and `FloatingPointLiteral` would produce incorrect results in some cases. That is, we should give developers a way to differentiate the two situations.

I think it'd also be worse including the underlying raw value type as an associated value of each case so that people can e.g. enforce that an enum's raw value is `UInt8` easily without having to repeat the parsing that MacroToolkit already does internally.